Assets/MixedRealityToolkit.Tests/PlayModeTests/NearInteractionTouchableTests.cs

```diff
@@ -421,6 +421,150 @@ public IEnumerator NearInteractionTouchableUnityUiToggle()
+        /// 
+        /// Default PokeThreshold is larger than the TouchableDistance, so we have to reduce it in the tests below to avoid dropping the active touchable before exceeding the DebounceThreshold.
+        /// 
+        /// The ray in the second step starts at Position - PokeThreshold, but this is not taken into account when raising events. So effectively one has to move the pointer right up to the collider in order to trigger a touch, regardless of the PokeThreshold. This may be intentional or not, it's not documented at all.
+        /// </remarks>
+        [UnityTest]
+        public IEnumerator NearInteractionTouchableDistance()
+        {
+            var touchableRect = CreateTouchable<NearInteractionTouchable>(0.15f);
+            touchableRect.SetLocalForward(touchNormal);
+            touchableRect.Bounds = new Vector2(0.5f, 0.5f);
+            var catcherRect = CreateEventCatcher(touchableRect);
+
+            var touchableVolume = CreateTouchable<NearInteractionTouchableVolume>(0.15f);
+            var catcherVolume = CreateEventCatcher(touchableVolume);
+
+            var canvas = UnityUiUtilities.CreateCanvas(0.002f);
+            var touchableUI = canvas.GetComponent<NearInteractionTouchableUnityUI>();
+            var button = UnityUiUtilities.CreateButton(Color.gray, Color.blue, Color.green);
+            button.transform.SetParent(touchableUI.transform, false);
+            var catcherUI = new UnityButtonEventCatcher(button);
+
+            yield return new WaitForFixedUpdate();
+            yield return null;
+
+            touchableRect.gameObject.name = "Rect Touchable";
+            touchableVolume.gameObject.name = "Volume Touchable";
+            touchableUI.gameObject.name = "UnityUI Touchable";
+
+            float distSurface = 0.3f;
+            touchableRect.transform.position = new Vector3(-1, 0, distSurface);
+            touchableVolume.transform.position = new Vector3(0, 0, distSurface);
+            touchableUI.transform.position = new Vector3(1, 0, distSurface);
+
+            yield return TestTouchableDistances(touchableRect, 0.0075f, touchableRect.gameObject);
+            yield return TestTouchableDistances(touchableVolume, 0.0075f, touchableVolume.gameObject);
+            yield return TestTouchableDistances(touchableUI, 0.0f, button.gameObject);
+
+            UnityEngine.Object.Destroy(touchableRect.gameObject);
+            UnityEngine.Object.Destroy(touchableVolume.gameObject);
+            UnityEngine.Object.Destroy(touchableUI.gameObject);
+        }
+
+        private IEnumerator TestTouchableDistances(BaseNearInteractionTouchable touchable, float colliderThickness, GameObject objectDownExpected)
+        {
+            Handedness handedness = Handedness.Right;
+            ArticulatedHandPose.GestureId gesture = ArticulatedHandPose.GestureId.Open;
+
+            yield return PlayModeTestUtilities.ShowHand(handedness, inputSim);
```

Comment:
you could use @luis-valverde-ms  hand utility class instead of this (and the other MoveHandFromTo calls)